### PR TITLE
📖Update documentation on layout intrinsic

### PR DESCRIPTION
**Pull request by cathyxz**
Follow up to https://github.com/ampproject/amphtml/issues/23453 and https://github.com/ampproject/amphtml/issues/24366.

### CODE REVIEW

File: builtins/amp-img.md [hunk 1]
```diff
@@ -164,7 +164,7 @@ If you want your image to scale as the window is resized but up to a maximum wid
...
+Both the `responsive` and `intrinsic` layouts create an image that will scale automatically.  The main difference is that the `intrinsic` layout uses an SVG image as its scaling element.  This will make it behave in the same way as a standard HTML image while retaining the benefit of the browser knowing the image size on initial layout. The `intrinsic` layout will have an intrinsic size and will inflate a floated `div` until it reaches either the natural image size (as defined by the width and height attributes passed to the amp-img, not by the natural size of the image) or a CSS constraint like `max-width`. The `responsive` layout will render 0x0 in a floated `div` because it takes its size from the parent, which has no natural size when floated.
```
Comment (CrystalOnScript) [inline builtins/amp-img.md]: I'd clarify this to:
...until it reaches either the image size defined by the `width` and `height` attributes passed to the `amp-img` (not by the natural size of the image) or a CSS constraint such as `max-width`.

File: spec/amp-html-layout.md [hunk 2]
```diff
@@ -122,7 +122,7 @@ Supported values for the `layout` attribute:
...
+      <td>The element takes the space available to it and resizes its height automatically to the aspect ratio given by the <code>width</code> and <code>height</code> attributes <em>until</em> it reaches the element's natural size (as defined by the given height and width attributes) or reaches a CSS constraint (e.g., max-width). The width and height attributes must be present. This layout works very well for most AMP elements, including <code>amp-img</code>, <code>amp-carousel</code>, etc. The available space depends on the parent element and can also be customized using <code>max-width</code> CSS. This layout differs from <code>responsive</code> by having an intrinsic height and width. This is most apparent inside a floated element where a <code>responsive</code> layout will render 0x0 and an <code>intrinsic</code> layout will inflate to the smaller of its natural size or any CSS constraint.</td>
```
Comment (CrystalOnScript) [inline spec/amp-html-layout.md]: Similar nit here:
...<em>until</em> it reaches the element's size defined by the `width` and `height` attributes passed to the `amp-img`, or reaches a CSS constraint, such as `max-width`.

File: builtins/amp-img.md (hunk 1 above)
Comment (cathyxz) [reply builtins/amp-img.md]: I'm glad. I was hoping we could make this explicit. =) I will make this change shortly.

**Pull request merged**